src/kernel/src/process/filedesc.rs

```diff
@@ -151,6 +150,36 @@ impl FileDesc {
     }
 }
 
+#[repr(C)]
+pub struct PollFd {
+    fd: i32,
+    events: PollEvents,
+    revents: PollEvents,
+}
+
+bitflags! {
+    #[repr(transparent)]
+    #[derive(Debug, Clone, Copy)]
+    pub struct PollEvents: u16 {
```

Comment:
This one on the PS4 is `u16`?

Reply:
in PollFd it's an i16 (so u16 is acceptable I think), but then the poll functions in cdevsw and fileops return an int just like all the other functions. This returned int(i32 for us) is then set into the i16 of the pollfd struct.

Reply:
If its size is inconsistent better to not use `repr(transparent)` here and put a warning in its documentation so people won't accept or pass this type directly between the PS4.

Reply:
There is no problem here. The only place this is received from the PS4 is in sys_poll and we are matching that size. All the other functions are internal and we just return PollEvents instead of and int. The higher bits would've been discarded anyway

Reply:
The problem here is this type used with `FileBackend::poll`, which represents an implementation of `fo_poll`. Once I saw this I will immediately assume this type has the same representation as the PS4 version because it has `#[repr(transparent)]`.

Reply:
Is it enough to just add a comment or you want something more?

Reply:
We need to remove `#[repr(transparent)]` and document this type that it cannot be passed directly from/to the PS4.

Reply:
But it can be passed, because it's the right size

Reply:
The point here is its size inconsistent. Yes it can be used directly in some functions. But some functions this type have different sizes.

Reply:
But again, why does it matter? All those functions that return a bigger size are purely internal and we can just return PollEvents and the behaviour will be the same. If all the information can always fit in a u16, why bother? We can just add a comment explaining our choice and then nothing bad can ever happen

Reply:
That was my intention